ext/standard/pack.c

```diff
@@ -979,6 +979,13 @@ PHP_FUNCTION(unpack)
 						zend_string *buf;
 						zend_long ipos, opos;
 
+
+						if (ZEND_LONG_INT_OVFL((zend_long)size * 2)) {
```

Comment:
On architectures where `SIZEOF_INT >= SIZEOF_ZEND_LONG` the macro evaluates to zero. However, the given reproducer might still work on 32bit architectures with [PAE](https://en.wikipedia.org/wiki/Physical_Address_Extension) enabled. Maybe a "manual" check is in order here:
```suggestion
						if (size > INT_MAX / 2)) {
```

Reply:
sure